ui/uihandler.go

```diff
@@ -962,37 +962,28 @@ func Report(baseurl, port, staticRsrcDir string, datadir string) bool {
 	return true
 }
 
-// setHostAndPort takes hostport in the form of hostname:port, ip:port or :port,
-// sets the urlHostPort variable.
+// setHostAndPort takes the bind address and sets urlHostPort as bind-address:port
+// unless the original input port didn't specify an address (starts with : or is just a port,
+// no colon) in which case it uses localhost:port
 func setHostAndPort(inputPort string, addr *net.TCPAddr) {
-	urlHostPort = inputPort
-	portStr := inputPort
-	if addr != nil {
-		urlHostPort = addr.String()
-		portStr = fmt.Sprintf(":%d", addr.Port)
-	}
-	if !strings.Contains(inputPort, ":") {
-		inputPort = ":" + inputPort
-	}
-	if strings.HasPrefix(inputPort, ":") {
-		urlHostPort = "localhost" + portStr
+	urlHostPort = addr.String()
+	if strings.HasPrefix(inputPort, ":") || !strings.Contains(inputPort, ":") {
+		urlHostPort = fmt.Sprintf("localhost:%d", addr.Port)
```

Comment:
Just to clarify - you expect `add.Port` to be the same as the provided `inputPort` (numerically)?

Reply:
inputPort could be a service name (e.g http or smtp etc...) that is used to bind and resolved internally by go but generally yes